midealocal/devices/da/__init__.py

```diff
@@ -137,30 +137,39 @@ def process_message(self, msg: bytes) -> dict[str, Any]:
         ]
         for status in self._attributes:
             if hasattr(message, str(status)):
+                value = getattr(message, str(status))
                 if status == DeviceAttributes.progress:
-                    self._attributes[status] = progress[getattr(message, str(status))]
+                    self._attributes[status] = (
+                        None if value >= len(progress) else progress[value]
+                    )
                 elif status == DeviceAttributes.program:
-                    self._attributes[status] = program[getattr(message, str(status))]
+                    self._attributes[status] = (
+                        None if value >= len(program) else program[value]
+                    )
                 elif status == DeviceAttributes.rinse_level:
-                    temp_rinse_level = getattr(message, str(status))
+                    temp_rinse_level = value
                     if temp_rinse_level == MIN_TEMP:
-                        self._attributes[status] = "-"
+                        self._attributes[status] = None
                     else:
                         self._attributes[status] = temp_rinse_level
                 elif status == DeviceAttributes.dehydration_speed:
-                    temp_speed = getattr(message, str(status))
-                    if temp_speed == MIN_TEMP:
-                        self._attributes[status] = "-"
-                    else:
-                        self._attributes[status] = speed[temp_speed]
+                    self._attributes[status] = (
+                        None if value >= len(speed) else speed[value]
+                    )
```

Comment:
Seems here we slightly changed logic for MIN_TEMP

Reply:
Should we return None or "-"?

Reply:
I already voted for "None", but if I get it right, now you return None only if value >= len(speed) and not also when equal to MIN_TEMP. Am I wrong ?